runtime/compiler/env/J9ClassEnv.cpp

```diff
@@ -417,24 +418,79 @@ J9::ClassEnv::isAnonymousClass(TR::Compilation *comp, TR_OpaqueClassBlock *clazz
    return comp->fej9()->isAnonymousClass(clazz);
    }
 
-const TR::TypeLayout*
-J9::ClassEnv::enumerateFields(TR::Region& region, TR_OpaqueClassBlock * opaqueClazz, TR::Compilation *comp)
-   {  
-   J9Class *clazz = (J9Class*)opaqueClazz;
-   TR_VMFieldsInfo fieldsInfo(comp, clazz, 1, stackAlloc);
-   ListIterator<TR_VMField> iter(fieldsInfo.getFields());
-   TR::TypeLayoutBuilder tlb(region);
-   for (TR_VMField *field = iter.getFirst(); field; field = iter.getNext())
+/*
+ * \param prefix
+ *    prefix could be ended with `.` or NULL.
+ * \param prefixLength
+ *    The length of the prefix string.
+ * \param toConcatenateAsPrefix
+ *    If true, the new name is appended with "." to be used as a part of the flattened field chain name.
+ * \param mergedLength
+ *    The merged length of the concatenated field name that is returned to the caller.
+ */
+static char * mergeFieldNames(char * prefix, uint32_t prefixLength, J9ROMFieldShape * field,
```

Comment:
Not a huge fan of the bool here - it is hard to read - an enum would be better or perhaps two functions with an impl using a boolean flag - I just don't like the boolean flags spraying all over the codebase....

Reply:
I'll use two functions. something like 
1. `mergeFieldNames` : simply merge names without appending . 
2. `buildTransitiveFieldNames` : merge names with `.` at the end.